osu.Game/Screens/SelectV2/BeatmapCarouselFilterMatching.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using osu.Game.Beatmaps;
+using osu.Game.Graphics.Carousel;
+using osu.Game.Screens.Select;
+
+namespace osu.Game.Screens.SelectV2
+{
+    public class BeatmapCarouselFilterMatching : ICarouselFilter
+    {
+        private readonly Func<FilterCriteria> getCriteria;
+
+        /// <summary>
+        /// The total number of beatmap difficulties displayed post filter.
+        /// </summary>
+        public int BeatmapItemsCount { get; private set; }
```

Comment:
Aren't you going to need this exposed at `BeatmapCarousel` to show on the filter control in the future? Would it make sense to expose it now and use that in tests while we're here?

Reply:
I've done exactly that in the filter control "# notes" PR (https://github.com/ppy/osu/pull/33048) but I can move that here.

Reply:
If it's not moved here, the tests using manual access will not be updated, I guarantee it.

And I can prove it:

https://github.com/ppy/osu/pull/33048/files#diff-dad761186b47d71ed7fa1a5078172207e62567ebb85efe945e9eb7460bcb475fR163-R170

Reply:
https://github.com/ppy/osu/pull/33046/commits/1c6e998c951de030da40e0f3389096ed9b1583df